packages/store/src/ngxs-error-handler.ts

```diff
@@ -0,0 +1,29 @@
+import { ErrorHandler, Injectable, Injector, NgZone } from '@angular/core';
+
+export interface NgxsErrorContext {
+  action?: any;
+}
+
+@Injectable({ providedIn: 'root' })
+export class NgxsUnhandledErrorHandler {
+  private _errorHandler: ErrorHandler = null!;
+
+  constructor(
+    private _ngZone: NgZone,
+    private _injector: Injector
+  ) {}
+
+  handleError(_error: any, _context: NgxsErrorContext): void {
+    // Retrieve lazily to avoid cyclic dependency exception.
+    this._errorHandler = this._errorHandler || this._injector.get(ErrorHandler);
```

Comment:
Given the other changes to load this lazily in the StateFactory, we should move this back to the constructor.
This would also be closer to what a user may implement.